extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-oauth2-adapter.acceptance.ts

```diff
@@ -0,0 +1,313 @@
+    status: number,
+  ) {
+    return new RedirectRoute('/', redirectUrl, status);
+  }
+
+  // we configure the callback url also with the same oauth2 strategy
+  @authenticate('oauth2')
+  // this SHOULD be a GET call so that the third party can ask the browser to redirect
+  @get('/auth/thirdparty/callback')
+  // thirdPartyCallBack() is the handler for '/auth/thirdparty/callback'
+  // the oauth2 strategy identifies this as a callback with the request.query.code sent by the third party app
+  // the oauth2 strategy exchanges the access code for a access token and then calls the provided verify() function
+  // the verify function creates a user profile after verifying the access token
+  thirdPartyCallBack(@inject(SecurityBindings.USER) user: UserProfile) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    return {access_token: user.token};
+  }
+}
+
+describe('Oauth2 authorization flow', () => {
+  let app: RestApplication;
+  let oauth2Strategy: StrategyAdapter<MyUser>;
+  let client: Client;
+
+  before(startMockOauth2Server);
+  after(stopMockOauth2Server);
+
+  before(givenLoopBackApp);
+  before(givenOauth2Strategy);
+  before(setupAuthentication);
+  before(givenControllerInApp);
+  before(givenClient);
+
+  let oauthProviderUrl: string;
+  let providerLoginUrl: string;
+  let callbackToLbApp: string;
+
+  context('Stage 1 - Authorization code grant', () => {
+    describe('when client invokes oauth flow', () => {
+      it('call is redirected to third party authorization url', async () => {
+        // HTTP status code 303 means see other, 
+        // on seeing which the browser would redirect to the other location
+        const response = await client.get('/auth/thirdparty').expect(303);
+        oauthProviderUrl = response.get('Location');
+        expect(url.parse(response.get('Location')).pathname).to.equal(
+          url.parse(oauth2Options.authorizationURL).pathname,
+        );
+      });
+
+      it('call to authorization url is redirected to oauth providers login page', async () => {
+        // HTTP status code 302 means redirect to new uri,
+        // on seeing which the browser would redirect to the new uri
+        const response = await supertest('').get(oauthProviderUrl).expect(302);
+        providerLoginUrl = response.get('Location');
+        expect(url.parse(response.get('Location')).pathname).to.equal('/login');
+      });
+
+      it('login page redirects to authorization app callback endpoint', async () => {
+        let params = url.parse(providerLoginUrl).query;
+        params = params + '&&username=user1&&password=abc';
+        // On successful login, the authorizing app redirects to the callback url
+        // HTTP status code 302 is returned to the browser
+        const response = await supertest('')
+          .post('http://localhost:9000/login_submit?' + params)
+          .send({username: 'user', password: 'abc'})
```

Comment:
I was having a hard time to submit the form with variables in the beginning, so tried all kinds of approach. I will remove the query parameters since form submission is working now.